### PR TITLE
fix unexpected right context merging causes recommednation gets trimmed

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Before
jb-before

## After
jb-after

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererRecommendationManager.kt [hunk 1]
```diff
@@ -138,7 +138,7 @@ class CodeWhispererRecommendationManager {
         val overlap =
             if (rightContextFirstLine.isEmpty()) {
                 val tempOverlap = overlap(recommendationContent, rightContext)
-                if (tempOverlap.isEmpty()) overlap(recommendationContent.trimEnd(), rightContext.trimStart()) else tempOverlap
+                if (tempOverlap.isEmpty()) overlap(recommendationContent.trimEnd(), rightContext) else tempOverlap
```
[the remark this answers is not in the document]
Comment (andrewyuq) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererRecommendationManager.kt:141]: Try to change it to only remove the newlines characters starting at the second character.

[the remark this answers is not in the document]
Comment (andrewyuq) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererRecommendationManager.kt:141]: and after the change try to test the cases when right context on the first line is not empty.

[the remark this answers is not in the document]
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererRecommendationManager.kt:141]: does this need tests?

[the remark this answers is not in the document]
Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererRecommendationManager.kt:141]: yea, let me add some

File: .changes/next-release/bugfix-71c6ba6c-d774-49b0-b7fb-81e14a1cb3f0.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "type" : "bugfix",
+  "description" : "CodeWhisperer suggestions may be trimmed under certain criteria"
```
Comment (rli) [inline .changes/next-release/bugfix-71c6ba6c-d774-49b0-b7fb-81e14a1cb3f0.json:3]:
```suggestion
  "description" : "Fix issue where CodeWhisperer suggestions are sometimes trimmed"
```

**Review by rli — APPROVED** [review]

**Pull request merged**